### PR TITLE
engineering: document error handling

### PR DESCRIPTION
**Pull request by smola**
Return-early on error is something we have been promoting very frequently
in Go projects, and to a lesser extend on other languages too.

Signed-off-by: Santiago M. Mola <santi@mola.io>

### CODE REVIEW

File: engineering/conventions.md [hunk 1]
```diff
@@ -47,6 +49,10 @@ We have additional conventions for each language:
 * Use latest [alpine](https://hub.docker.com/_/alpine/) or [busybox](https://hub.docker.com/_/busybox/) whenever it is possible. Use latest [debian](https://hub.docker.com/_/debian/) stable-slim otherwise.
 * If needed to reduce image size, use [multi-stage builds](https://docs.docker.com/develop/develop-images/multistage-build/).
 
+## Error Handling
+
+* If it is natural in your language, prefer early handling of errors and return early (more info: [here](http://blog.timoxley.com/post/47041269194/avoid-else-return-early), [here](http://wiki.c2.com/?GuardClause) and [here](http://wiki.c2.com/?HandleErrorsInContext)).
```
Comment (campoy) [inline engineering/conventions.md]: s/natural/idiomatic/

Comment (campoy) [reply engineering/conventions.md]: I'd replace the `here` by something more meaningful.

More info:
- [Avoid else, return early](http://blog.timoxley.com/post/47041269194/avoid-else-return-early)
etc

**Comment by smola** [comment]
I put this on hold, since @mcuadros is covering error handling in his update about logging.

**Comment by smola** [comment]
Merging, since this is well-established practice and no contentious. There's no point in leaving this on hold.

**Pull request merged**